presto-main/src/test/java/io/prestosql/tests/LogTestDurationListener.java

```diff
@@ -117,10 +121,25 @@ else if (finished.get()) {
 
     private static void dumpAllThreads(String message)
     {
-        LOG.warn("%s\n\nFull Thread Dump:\n%s", message,
-                Arrays.stream(getThreadMXBean().dumpAllThreads(true, true))
-                        .map(ThreadInfo::toString)
-                        .collect(joining("\n")));
+        ListMultimap<String, ThreadInfo> threadInfos = Multimaps.index(
+                Arrays.asList(getThreadMXBean().dumpAllThreads(true, true)),
+                threadInfo -> Arrays.toString(threadInfo.getStackTrace()));
+        LOG.warn("%s\n\nThread Dump:\n%s", message,
+                threadInfos.keySet().stream()
+                        .map(threadStack -> {
+                            List<ThreadInfo> similarThreads = threadInfos.get(threadStack);
```

Comment:
iterate over `entrySet()` or `asMap().entrySet()`

Reply:
`entrySet()` unrolls the multimap